lib/rest2.js

```diff
@@ -220,6 +220,9 @@ class RESTv2 {
    * @private
    */
   async _makePublicRequest (path, cb, transformer) {
+    if (cb !== null && typeof cb !== 'function') {
+      throw new Error('callback must be a function')
+    }
```

Comment:
I have a doubt: we are checking if cb is valid but in other request functions we don't ... do you think this function should check the cb and the others not ? or why we are checking this here?

Reply:
I added this check because my test was hanging when passing a params object instead of a callback in this function.

Reply:
hmm I still don't understand this, where are we passing object to cb?

Reply:
It is just defensive programming, so if somebody pass the wrong argument to the function you notice.

There was during development a moment when I wrote:
```js
const params = {}
this._makePublicRequest('/some/endpoint', params, cb)
```

and the tests got stuck because cb was not being called, so I wanted to add some hints, I can revert this if you want.

this,mak

Reply:
ok can you adjust it to support nil (undefined + null)?